src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,405 @@
+
+        private void CreateAndCommitAuthSettings(string hostName, string clientId, string clientSecret, string tenant, List<string> replyUrls)
+        {
+            // The WEBSITE_AUTH_ALLOWED_AUDIENCES setting is of the form "{replyURL1} {replyURL2}"
+            string serializedReplyUrls = string.Join(" ", replyUrls);
+
+            // Create a local auth .json file that will be used by the middleware
+            var middlewareAuthSettingsFile = SecretsManager.MiddlewareAuthSettingsFileName;
+            MiddlewareAuthSettings = new AuthSettingsFile(middlewareAuthSettingsFile);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ALLOWED_AUDIENCES", serializedReplyUrls);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_AUTO_AAD", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_CLIENT_ID", clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_CLIENT_SECRET", clientSecret);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_DEFAULT_PROVIDER", "AzureActiveDirectory");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENABLED", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_OPENID_ISSUER", "https://sts.windows.net/" + tenant + "/");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_RUNTIME_VERSION", "1.0.0");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_TOKEN_STORE", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_UNAUTHENTICATED_ACTION", "AllowAnonymous");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_HOSTNAME", hostName);
+
+            // Create signing and encryption keys for local testing
+            string encryptionKey = ComputeSha256Hash(clientSecret);
+            string signingKey = ComputeSha256Hash(clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENCRYPTION_KEY", encryptionKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_SIGNING_KEY", signingKey);
+            MiddlewareAuthSettings.Commit();
+
+            ColoredConsole.WriteLine(Yellow($"Created {SecretsManager.MiddlewareAuthSettingsFileName} with authentication settings necessary for local development.\n" +
+                $"Running this function locally will only work with the Function Host default port of {StartHostAction.DefaultPort}"));
+
+            if (_workerRuntime == WorkerRuntime.dotnet)
+            {
+                // If this is a dotnet function, we also need to add this file to the .csproj 
+                // so that it copies to \bin\debug\netstandard2.x when the Function builds
+                string csProjFile = GetCSProjFilePath();
+
+                if (csProjFile == null)
+                {
+                    throw new CliException($"Auth settings file {SecretsManager.MiddlewareAuthSettingsFileName} could not be added to a .csproj and will not be present in the the bin or output directories.");
```

Comment:
C# scripts (.csx) are .NET, but they don't have a .csproj file. Will this exception exclude those users? I wonder if it's safer to just skip that step and write an informational trace if we can't find the .csproj file.